### PR TITLE
Storybook: adjust arg table of ReferenceLine

### PR DESCRIPTION
**Pull request by lukask-proxora**
## Related Issue

#3433 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: storybook/stories/API/cartesian/ReferenceLine.stories.tsx [hunk 1]
```diff
@@ -67,13 +99,6 @@ export default {
       disable: true,
       default: false,
     },
-    // Other
-    segment: {
-      description: 'Array of endpoints in { x, y } format. These endpoints would be used to draw the ReferenceLine.',
-      table: { type: { summary: 'array' }, category: 'Other' },
-    },
-    // Internal
-    data: { table: { category: 'Internal' } },
```
Comment (nikolasrieble) [inline storybook/stories/API/cartesian/ReferenceLine.stories.tsx:76]: It seems this prop is now missing. Why?

File: storybook/stories/API/cartesian/ReferenceLine.stories.tsx [hunk 2]
```diff
@@ -47,16 +57,38 @@ const StyleProps: Args = {
...
+    table: {
+      type: {
+        summary: 'ReactElement | Function',
+      },
+      category: 'Style',
+    },
+  },
+};
+
+const InternalProps: Args = {
```
Comment (nikolasrieble) [inline storybook/stories/API/cartesian/ReferenceLine.stories.tsx:77]: From the type I see further InternalProps, such as 

```
interface InternalReferenceLineProps {
  viewBox?: CartesianViewBox;
  xAxis?: Omit<XAxisProps, 'scale'> & { scale: D3Scale<string | number> };
  yAxis?: Omit<YAxisProps, 'scale'> & { scale: D3Scale<string | number> };
  clipPathId?: number | string;
}
```

Should they also be added?

**Review by ckifer — APPROVED** [review]
Looks good for now either way, thanks

File: storybook/stories/API/cartesian/ReferenceLine.stories.tsx (hunk 2 above)
Comment (lukask-proxora) [reply storybook/stories/API/cartesian/ReferenceLine.stories.tsx:77]: Hi @nikolasrieble, the remaining internal props are added through `ReferenceComponentInternalArgs`

**Pull request merged**